run-build-functions.sh

```diff
@@ -685,9 +689,13 @@ cache_artifacts() {
   cache_home_directory ".boot" "boot dependencies"
   cache_home_directory ".composer" "composer dependencies"
   cache_home_directory ".homebrew-cache", "homebrew cache"
-  cache_home_directory ".rustup" "rust rustup cache"
-  cache_home_directory ".cargo/registry" "rust cargo registry cache"
-  cache_home_directory ".cargo/bin" "rust cargo bin cache"
+  
+  if [ -f Cargo.toml ] || [ -f Cargo.lock ]
+  then
+    cache_home_directory ".rustup" "rust rustup cache"
```

Comment:
I think we should try to cache this regardless of whether there are Cargo config files

Reply:
updated